src/services/db/__tests__/GitHubService.spec.ts

```diff
@@ -132,12 +151,88 @@ describe("Github Service", () => {
         sha,
       })
       expect(mockAxiosInstance.put).toHaveBeenCalledWith(
-        endpoint,
+        folderEndpoint,
         params,
         authHeader
       )
     })
 
+    it("Creating a top level folder works correctly", async () => {
+      const collectionYmlMessage = JSON.stringify({
+        message: `Create file: ${topLevelDirectoryFileName}`,
+        fileName: topLevelDirectoryFileName,
+        userId,
+      })
+
+      const resp = {
+        data: {
+          content: {
+            sha,
+          },
+        },
+      }
+      mockAxiosInstance.put.mockResolvedValueOnce(resp)
+      mockAxiosInstance.get.mockResolvedValueOnce("")
+      await expect(
+        service.create(sessionData, {
+          content: collectionYmlContent,
+          fileName: topLevelDirectoryFileName,
+          directoryName,
+        })
+      ).resolves.toMatchObject({
+        sha,
+      })
+      const resourceRoomParams = {
+        message: collectionYmlMessage,
+        content: Base64.encode(collectionYmlContent),
+        branch: params.branch,
+      }
+      const topLevelFolderEndpoint = `${folderParentEndpoint}/${topLevelDirectoryFileName}`
```

Comment:
nit: could we shift this to be grouped with the other arrange stuff (eg: initialising `resp`)